Fixed separator between SuperPins and normal tabs.

CosmoCreeper opened the pull request:
I propose a solution to the separator issue between SuperPins and normal tabs. My solution utilizes grid-column on the separator.

CosmoCreeper commented:
Thx for the support, I just updated the title as I have also fixed the bleeding of pinned tabs from other workspaces. (Becuase I fixed it, I also removed the visibility trick that the devs were working on.)

JLBlk commented:
Hey, thanks a lot for all your hard work. The pr looks great, just one tiny thing: when having 2 or more pinned tabs  and the tab bar is expanded enough, the pins don't span across the whole row

![image](https://github.com/user-attachments/assets/6af9576b-1bcd-4efd-9316-1704bf15deee)

I unfortunately couldn't find a fix for that, maybe you have an idea (you seem more experienced than me).
I'll merge it either way if you find a fix or not since it's just an edge case and everything else seems great.
Thanks again for your work ❤️.

TODO: Add credits in the readme.

CosmoCreeper commented:
Man, I really thought that this was a part of the original code. I'll see if I can find a fix but I'm also working on another feature so be expecting a pull request soon!

CosmoCreeper commented:
Turns out that is just a side effect of grid-column and I can't really do much about that (alternative is to absolutely position it but that has problems with the separator styling.)

Side note: I just implemented #55 for pinned tabs as well as an option for it, couldn't do it for Essentials just yet because of some styling issues with the way Zen does stuff. (particularly a min-width: fit-content on the sidebar) I just kept the logic for adding this feature for Essentials but there is no option to do so and it is just there as a WIP. (Just as a refresher, #55 is covering the full width when only one tab is open on its row.)

https://github.com/user-attachments/assets/c49ff1a7-9cdc-472f-98cb-06561dc254c9

CosmoCreeper commented:
Okay, apparently there was an issue with setting the Essentials width to be the same on Pinned Tabs if auto-grow is on. (It wouldn't acutally apply the width because it wasn't being applied on the tab and rather on the workspace pinned tabs area.) I fixed this in a new commit and also added a new feature. I switched the dim unloaded tabs to a dropdown and added the ability to dim text, icons, or a combination of the two. (I decided to add this feature because of the issue, #51.)

CosmoCreeper commented:
> ![image](https://private-user-images.githubusercontent.com/118555638/433961655-6af9576b-1bcd-4efd-9316-1704bf15deee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ3NjE0NzEsIm5iZiI6MTc0NDc2MTE3MSwicGF0aCI6Ii8xMTg1NTU2MzgvNDMzOTYxNjU1LTZhZjk1NzZiLTFiY2QtNGVmZC05MzE2LTE3MDRiZjE1ZGVlZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxNVQyMzUyNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZmQwYWFhNzhkYWE4N2UwMzBiYjQwZDM5NjEyNWRiMjEwZDU3NTcwN2E5Yjc1MWNhZmNkMTdlN2FlNzViNzUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KWloo1YuuEdt1xsYM6RuJ7SFTTJtnj6H4UbMb8bZKcc)

I just updated the pull request with a new version that fixes this issue by changing the separator styling from grid-column to position: absolute. I also noticed a bug that would result in the separator being incredibly thin due to the gap property. I fixed this by removing the gap property and replacing it with margin-right and margin-bottom.

JLBlk commented:
Damn you a wizard lol. The growth feature looks really amazing. Remember to also commit the preferences.json changes.

CosmoCreeper commented:
Okay, just did the preferences. I set the default dimming of unloaded tabs to be both icon and text but it doesn't have to be that way. Would you like it to be icons only by default or is what I have fine?

CosmoCreeper commented:
I just added bottom essentials to the settings. #60 (Looks pretty good.)

![Screenshot 2025-04-16 122732](https://github.com/user-attachments/assets/2cd03ec4-ee19-4a7f-81ff-81a9aa47393b)

CosmoCreeper commented:
With the new commit, when the user has "zen.tabs.dim-pending" on and adjusts the dim settings, the icons will always be dimmed no matter what.

CosmoCreeper commented:
I just created a new commit that overwrites the built-in dim-pending.

CosmoCreeper commented:
Yeah I think I fixed everything I wanted to.

JLBlk commented:
Nice, thank you for all your hard work ❤️ ! Your really elevated this mod to another level

JLBlk commented:
I'll do the pr now to update the mod officially for everyone 👍

CosmoCreeper commented:
Okay. Also make sure to close all of those issues that are fixed now.

JLBlk commented:
Yeah, i'll do that ones the pr for the official store gets merged. Thanks again for all your work!
